app/services/articles/feeds/weighted_query_strategy.rb

```diff
@@ -0,0 +1,398 @@
+module Articles
+  module Feeds
+    # @api private
+    #
+    # This is an experimental object that we're refining to be a
+    # competetor to the existing feed strategies.
+    #
+    # @note One possible short-coming is that the query does not
+    #   account for the Forem's administrators.
+    # @note For those considering extending this, be very mindful of
+    #   SQL injection.
+    class WeightedQueryStrategy
+      # This constant defines the allowable relevance scoring methods.
+      #
+      # A scoring method should be a SQL fragement that produces a
+      # value between 0 and 1.  The closer the value is to 1, the more
+      # relevant the article is for the given user.  Note: the values
+      # are multiplicative.  Make sure to consider if you want a 0
+      # multiplier for your score.
+      #
+      # In addition, as part of initialization, the caller can
+      # configure each of the scoring methods :cases and :fallback.
+      #
+      # Each scoring method has three keys:
+      #
+      # - clause: The SQL clause statement; note: there exists a
+      #           coupling between the clause and the SQL fragments
+      #           that join the various tables.
+      #
+      # - cases: An Array of Arrays, the first value is what matches
+      #          the clause, the second value is the multiplicative
+      #          factor.
+      #
+      # - fallback: When no case is matched use this factor.
+      #
+      # - requires_user: Does this scoring method require a given
+      #                  user.  If not, don't use it if we don't have
+      #                  a nil user.
+      SCORING_METHOD_CONFIGURATIONS = {
+        # Weight to give based on the age of the article.
+        daily_decay_factor: {
+          clause: "(current_date - published_at::date)",
+          cases: [
+            [0, 1], [1, 0.95], [2, 0.9],
```

Comment:
These are a lot of cases, we could use a relatively simple calculation of `1 - 0.5 * days`. The only downside is that it's a bit more linear than what you currently have, though from what I see the only values changing would be 11, 12, and 13. 

This makes me wonder if an array is really the best/only possible type for `cases`. Maybe it could be lambda instead that receives what is currently the first array element as an argument?

Again, this is probably way past the scope of this PR, I'm just trying to envision where this could go in the future.

Reply:
I originally had a `1 - 0.5 * days` but when I moved to the configuration, it was the "proud nail" (e.g. the one that was different from the others).  I had also considered a lambda instead but favored the literal version for this go around.

Assuming the weighted strategy class is a reasonable contender against the other current strategy I'm prepared to invest more energy into its implementation.

One thought I definitely had was moving these cases into the database so that admins could completely tweak the structure.  Perhaps upload a JSON file that encapsulates the cases and fallback while also validating the JSON via a schema assertion.

Reply:
Fair, like I said I was primarily trying to anticipate where this could eventually go, totally fine to leave this as-is for now 😃